### PR TITLE
[Feature Add]:Added Mac OS Support for Edge and Opera

### PR DESCRIPTION
**Pull request by abhisinha2001**
# Description

Added Mac OS path for Microsoft Edge and Opera browser, thereby adding support for Mac OS in these browsers.

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# Checklist:

- [X] I have read the [contribution guidelines](https://browser-history.readthedocs.io/en/latest/contributing.html) and followed it as far as possible. 
- [X] I have performed a self-review of my own code (if applicable)
- [ ] I have commented my code, particularly in hard-to-understand areas (if applicable)
- [ ] I have made corresponding changes to the documentation (if applicable)
- [X] My changes generate no new warnings 
- [X] Any dependent and pending changes have been merged and published

### CODE REVIEW

**Review by sujanchegu — APPROVED** [review]
I think someone with a Mac might have to check the actual path & the working. Rest seems good :smile:

**Comment by abhisinha2001** [comment]
I had used my Mac to check the path and did perform a couple a test cases locally. The changes made worked and it didn't have any conflicts as well😅

File: browser_history/browsers.py [hunk 1]
```diff
@@ -106,18 +106,39 @@ class Safari(Browser):
...
+
+    * Mac OS
+
+    Profile support: Yes
+    """
+    name = "Opera"
+    mac_path = 'Library/Application Support/com.operasoftware.Opera/'
+
+    profile_support = True
+    profile_dir_prefixes = Chrome.profile_dir_prefixes
```
Comment (Samyak2) [inline browser_history/browsers.py:140]: According to #41, this does not work (i.e., opera does not have profile support).
Did it work for you?

File: browser_history/browsers.py [hunk 2]
```diff
@@ -34,15 +34,17 @@ class Chrome(Browser):
...
 
     * Linux
+    * Windows
 
     Profile support: Yes
     """
     name = "Chromium"
 
     linux_path = '.config/chromium'
+    windows_path = 'AppData/Local/chromium/User Data'
```
Comment (Samyak2) [inline browser_history/browsers.py:47]: The PR title says "Mac OS support for Edge and Opera" but your changes are for Chromium on Windows :sweat_smile:?

**Review by Samyak2 — CHANGES_REQUESTED** [review]

File: browser_history/browsers.py (hunk 2 above)
Comment (abhisinha2001) [reply browser_history/browsers.py:47]: I had not touched the code for chromium browser and had added the required paths for Mac OS in Edge and Opera.
Will resolve this issue as soon as possible. I am facing some issues with merging the repo hence there might have some issues due to that.

File: browser_history/browsers.py (hunk 1 above)
Comment (abhisinha2001) [reply browser_history/browsers.py]: As such I did not encounter any issues with profile support and it worked without any hitches.

File: browser_history/browsers.py (hunk 2 above)
Comment (Samyak2) [reply browser_history/browsers.py:47]: Right, I noticed that this happened when merging with master. You could try one of these methods:
 - copying the changes from your [first commit](https://github.com/pesos/browser-history/pull/44/commits/1369a04176652d7f1e877e243e7fbda1c9ae89a0) and committing them again
 - [interactive rebase](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase) on `upstream master` and drop the merge commits
 - delete your branch, close this PR, make your changes again and open a new PR

Comment (abhisinha2001) [reply browser_history/browsers.py:47]: Thank you very much! Will try and resolve by tonight.

File: browser_history/browsers.py [hunk 3]
```diff
@@ -33,11 +29,8 @@ class Chrome(Browser):
 
 class Chromium(Browser):
     """Chromium Browser
-
     Supported platforms (TODO: Windows and Mac OS support)
-
     * Linux
-
```
Comment (Samyak2) [inline browser_history/browsers.py:40]: Were these by mistake?

File: browser_history/browsers.py [hunk 4]
```diff
@@ -124,38 +110,16 @@ class Edge(Browser):
...
-    Profile support: No
-    """
-    name = 'OperaGX'
-
-    windows_path = 'AppData/Roaming/Opera Software/Opera GX Stable'
-
-    profile_support = False
-
-    history_file = Chrome.history_file
-    history_SQL = Chrome.history_SQL
```
Comment (Samyak2) [inline browser_history/browsers.py:161]: You accidentally reverted a feature added by another PR :sweat_smile:

**Review by Samyak2 — CHANGES_REQUESTED** [review]
Hey, looks like your branch got messed up from the merge. I would suggest you make a clean fork again and send in a new PR.

File: browser_history/browsers.py (hunk 4 above)
Comment (abhisinha2001) [reply browser_history/browsers.py:161]: Alright 
Will make a new PR then

**Pull request closed without merging**